app/views/spree/admin/locales/show.html.erb

```diff
@@ -1,27 +1,59 @@
 <%= render 'spree/admin/shared/configuration_menu' %>
 
-<%= form_tag admin_locale_path, method: :patch do %>
-  <div class="panel panel-default panel-localization">
-    <div class="panel-heading">
-      <h1 class="panel-title">
-        <%= Spree.t(:'i18n.localization_settings') %>
-      </h1>
-    </div>
-
-    <div class="panel-body" data-hook="localization_settings_body">
-      <div class="form-group">
-        <label for="available_locales_">
-          <%= Spree.t(:'i18n.available_locales') %>
-        </label>
-        <%= select_available_locales %>
-        <p class="help-block">
-          <%= Spree.t(:'i18n.locales_displayed_on_frontend_select_box') %>
-        </p>
-      </div>
-    </div>
-  </div>
-
-  <div class="form-buttons filter-actions actions" data-hook="buttons">
-    <%= button Spree.t('actions.update') %>
-  </div>
+<% content_for :page_title do %>
+  <%= Spree.t(:'i18n.available_locales') %>
 <% end %>
+
+<table class="index locales-settings">
+  <thead>
+    <tr>
+      <th>Name</th>
+      <th>Url</th>
+      <th><%= Spree.t(:'i18n.locales_displayed_on_frontend_select_box') %></th>
+      <th class="actions"></th>
+    </tr>
+  </thead>
+  <tbody>
+    <% Spree::Store.all.each do |store| %>
+      <%= form_tag admin_locale_path, method: :patch do %>
+        <tr class='edit-available-locales hidden store-id-<%= store.id %>'>
+          <td>
+            <%= store.name %>
+            <% if store.default? %>
+              <span class="label label-default">default</span>
+            <% end %>
+          </td>
+          <td><%= store.url %></td>
+          <td>
+            <%= select_available_locales(store) %>
+            <p class="help-block">
+            </p>
+          </td>
+          <td class="actions">
+            <%= link_to '', '#', :class => 'fa fa-check save-available-locales no-text with-tip', :data => {store_id: store.id, action: :save}, :title => Spree.t('save') %>
+            <%= link_to '', '#', :class => 'fa fa-cancel cancel-available-locales no-text with-tip', :data => {store_id: store.id, action: :cancel}, :title => Spree.t('cancel') %>
+          </td>
+        </tr>
+        <tr class='show-available-locales store-id-<%= store.id %>'>
+          <td>
+            <%= store.name %>
+            <% if store.default? %>
+              <span class="label label-default">default</span>
+            <% end %>
+          </td>
+          <td><%= store.url %></td>
+          <td>
+            <span id="available-locales-presentation-text-<%= store.id %>">
+              <%= available_locales_presentation(store) %>
+            </span>
+            <p class="help-block">
+            </p>
+          </td>
+          <td class="actions">
+            <%= link_to '', '#', :class => 'fa fa-edit edit-available-locales icon_link icon-edit no-text with-tip', :data => {store_id: store.id, action: 'edit'}, :title => Spree.t('edit') %>
+          </td>
+        </tr>
+      <% end %>
```

Comment:
I think a handlebars template would be much better here. 

Very sad we can't make use of `editable-table` introduced in solidusio/solidus/pull/1580 here, because of back wards compatibility, but maybe the same implementation can be used here?